services/cypress/integration/cmsWorkflow/rateReview.spec.ts

```diff
@@ -29,49 +29,54 @@ describe('CMS user can view rate reviews', () => {
-                const rate2 = submission.rateInfos[1]
-                cy.get('table')
-                .findByRole('link', { name: rate1.rateCertificationName })
-                .should('exist')
-                cy.get('table')
-                .findByRole('link', { name: rate2.rateCertificationName })
-                .should('exist')
+                (contract) => {
+
+                    const latestSubmission = contract.packageSubmissions[0]
+
+                    const rate1 = latestSubmission.rateRevisions[0]
+                    const rate2 = latestSubmission.rateRevisions[1]
+                    let rate1Name = rate1.formData.rateCertificationName
+                    let rate2Name = rate2.formData.rateCertificationName
+
+                    if (!rate1Name || !rate2Name) {
+                        throw new Error(`Unexpected error: Rate name(s) did not exist. Rate1Name: ${rate1Name}, Rate2Name: ${rate2Name}`)
+                    }
 
-                // click the first rate to navigate to rate summary page
-                cy.get('table')
-                .findByRole('link', { name: rate1.rateCertificationName })
-                .should('exist').click()
-                cy.url({ timeout: 10_000 }).should('contain',rate1.id)
-                cy.findByRole('heading', {
-                    name: `${rate1.rateCertificationName}`,
-                }).should('exist')
-                cy.findByText('Rate certification type').should('exist').siblings('dd').should('have.text', 'New rate certification')
-                cy.findByText('Rating period').should('exist').siblings('dd').should('have.text', '06/01/2025 to 05/30/2026')
-                cy.findByText('Date certified').should('exist').siblings('dd').should('have.text', '04/15/2025')
-                cy.findByText('Submission this rate was submitted with').should('exist').siblings('dd').should('have.text', submissionName)
-                cy.findByText('Certifying actuary').should('exist').siblings('dd').should('have.text', 'actuary1test titleemail@example.comMercer')
-                // cy.findByText('Download all rate documents').should('exist')
-                cy.findByRole('table', {
-                    name: 'Rate certification',
-                }).should('exist')
-                cy.findByText('rate1Document1.pdf').should('exist')
-                cy.findByRole('table', {
-                    name: 'Rate supporting documents',
-                }).should('exist')
+                    // Then check both rates in rate reviews table
+                    cy.logInAsCMSUser({
+                        initialURL: `/dashboard/rate-reviews`,
                     })
+
+                    // Rate names can be the same, rare in prod, but common in automated tests and manual tests.
+                    // Here were just checking to make sure the first exists in findAll.
+                    cy.get('table')
+                    .findAllByRole('link', { name: rate1Name }).first()
```

Comment:
I would use a data-testid for this with a unique identifier  (maybe the rate number). this feels like it could cause us issues down the road if we selecting by non unique value